frontend/homework/13/App.tsx

```diff
@@ -0,0 +1,93 @@
+
+import { useEffect, useState } from 'react';
+import './App.scss';
+
+import { ProductInfo } from './ProductInfo';
+import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
+import ProductList from './ProductListComponent';
+import HeaderComponent from './HeaderComponent';
+import Heading from './HeadingComponent';
+import ProductDetailsViewComponent from './ProductDetailsViewComponent';
+
+function App() {
+  const [allProductinfo, setAllProductinfo] = useState<ProductInfo[]>([]);
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [selectedBrand, setSelectedBrand] = useState<string>('All');
+  const [sortOrder, setSortOrder] = useState<string>('asc');
+  const [showProductList, setShowProductList] = useState<boolean>(true);
+
+  const fetchData = () => {
+    fetch('https://fakestoreapi.com/products')
+      .then((response) => response.json())
+      .then((data: ProductInfo[]) => {
+        setAllProductinfo(data);
+      });
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm(event.target.value);
+  };
+
+  const handleBrandChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedBrand(event.target.value);
+  };
+
+  const handleSortChange = () => {
+    setSortOrder((prevOrder) => (prevOrder === 'asc' ? 'desc' : 'asc'));
+  };
+
+  const filteredProducts = allProductinfo.filter((product) => {
+    const titleMatch = product.title.toLowerCase().includes(searchTerm.toLowerCase());
+    const brandMatch = selectedBrand === 'All' || product.category.toLowerCase() === selectedBrand.toLowerCase();
+    return titleMatch && brandMatch;
+  });
+
```

Comment:
code is too cluttered. Use design patterns like ContainerPresentation Design pattern or break it down into more components.